samples/server/go-server/api/server/storage/queries.go

```diff
@@ -16,48 +16,48 @@ package storage
 
 const (
 	createTables = `
-		CREATE TABLE IF NOT EXISTS project (
+		CREATE TABLE IF NOT EXISTS projects (
 			id SERIAL PRIMARY KEY,
 			name TEXT NOT NULL UNIQUE
 		);
-		CREATE TABLE IF NOT EXISTS note (
+		CREATE TABLE IF NOT EXISTS notes (
 			id SERIAL PRIMARY KEY,
 			name TEXT NOT NULL UNIQUE,
 			data TEXT
 		);
-		CREATE TABLE IF NOT EXISTS occurrence (
+		CREATE TABLE IF NOT EXISTS occurrences (
```

Comment:
It might be easier here for easier querying to pull 

note_project
note_name 
 
into separate columns. This allows things like listNoteOccurrences to be a bit easier to query.

Reply:
This would change the behaviour. The question is if we want an occurrence to change when its note is changed? 

At this point we serialise the complete occurrence, so unless we change this as well we probably don't want the occurrence to be updated when the note is updated. OTOH, not doing this change could lead to potential data integrity issues.

Sidenote: I personally think we can live with a query that is a bit more complex if it does what we want it to do.

Reply:
@R2wenD2 thoughts?

Reply:
We want to join against the notes table for the fields that are shared between the note and occurrence, so that we don't need to update *all* occurrences every time a note changes. In the long run, the join should be cheaper than the update.
wdyt?

(also, apologies for the delayed responses).

Reply:
Ok. So then we don't pull `note_project` and `note_name` into separate columns (per initial comment) and instead keep `note_id int REFERENCES notes` and join with notes table to get note_project and note_name. Right? SGTM but goes agains the initial comment.

Reply:
I have a mental model where note is keyed on (projectId, note name) and not note_id field. That is definitely not what's in the schema though :) 

Yeah, what you're suggesting works, lets do that.

Reply:
Have a feeling that a higher bandwidth communication channel than a PR review would make it easier to come to a shared understanding on this 😄

Anyway. Since it sounds like the solutions are (more or less) equivalent I'll go ahead with my suggestion. We can always revisit...